zerver/actions/realm_settings.py

```diff
@@ -186,13 +186,26 @@ def do_set_realm_message_editing(
     send_event(realm, event, active_user_ids(realm.id))
 
 
-def do_set_realm_notifications_stream(
-    realm: Realm, stream: Optional[Stream], stream_id: int, *, acting_user: Optional[UserProfile]
+def do_set_realm_stream(
+    realm: Realm,
+    field: str,
```

Comment:
It may be useful to type this as a `Literal["notifications_stream", "signup_notifications_stream"]` -- and possibly add an assert in an `else` for your `field` checks, below, to perform [exhaustiveness-checking](https://mypy.readthedocs.io/en/stable/literal_types.html#exhaustiveness-checking).

Reply:
Yeah, that's a good idea.